src/qamqpqueue_p.h

```diff
@@ -19,6 +22,48 @@ class QAmqpQueuePrivate: public QAmqpChannelPrivate,
         METHOD_ID_ENUM(miDelete, 40)
     };
 
+    /*!
+     * A class representing the state of this queue's subscriptions to
+     * a given exchange.
+     */
+    class SubscriptionState {
+    public:
+        /*! Exchange pointer, for checking state */
+        QPointer<QAmqpExchange> exchange;
```

Comment:
I can split these up, no problems there.

What the exchange pointer does there is it tells the state machine what exchange/topic was last being bound/unbound.  I initially went looking for this information in the queue.bind-ok and queue.unbind-ok messages but discovered they don't actually state what was bound or unbound, just that the last bind/unbind succeeded.

So to know which has been bound/unbound, I record it just prior to issuing the instruction, then when the response comes back, I know what got processed.  It could be an exchange name too I suppose, this would work equally well.